### ISSUE TITLE
Apps - Backend: Validation layer 

### ISSUE THREAD

**Issue by fabrizzio-dotCMS** [dotCMS/core#18846]
As a consumer of the apps module, I want to be able to have a flexible way to register custom validators that will allow me to write specific validations for my secrets. 
The validations will be performed by subscribing listeners to a "Save App Event"
We desire to have basic validation tools for fields that might hold JSON or XML type of values but their use is up to the specific validator code.  
The Save Apps Event Listener will be responsible for broadcasting an Asynchronous event that will inform about the validation results to the UI through the WebSocket Notifications.
In conclusion, the Apps module will only be responsible for broadcasting the "Save App Event" 
Users can still save secrets even though they are faulty or invalid. Any validation error message will be delivered as a system notification.

**Comment by fabrizzio-dotCMS** [comment]
PR https://github.com/dotCMS/core/pull/18883

**Comment by nollymar** [comment]
Passed internal QA. Tested on SAML integration

**Comment by bryanboza** [comment]
Fixed, tested on master // Postgres // FF
